File: adapter.cpp
#include <iostream>
#include <string>

class InternalClass {
public:
    std::string fetch() {
        return "get user info";
    }
};

class ExternalClass {
public:
    std::string search() {
        return "get user info";
    }
};

// 어댑터 클래스
class Adapter {
private:
    ExternalClass* external;

public:
    Adapter(ExternalClass* ext) : external(ext) {}

    std::string fetch() {
        return external->search();
    }
};

int main() {
    ExternalClass ext;
    Adapter adapter(&ext);

    std::cout << adapter.fetch() << std::endl;  // get user info
    return 0;
}

File: adapter.cs
using System;

class InternalClass {
    public string Fetch() {
        return "get user info";
    }
}

class ExternalClass {
    public string Search() {
        return "get user info";
    }
}

// 어댑터 클래스
class Adapter {
    private ExternalClass external;

    public Adapter(ExternalClass ext) {
        external = ext;
    }

    public string Fetch() {
        return external.Search();
    }
}

class Program {
    static void Main() {
        ExternalClass ext = new ExternalClass();
        Adapter adapter = new Adapter(ext);

        Console.WriteLine(adapter.Fetch());  // get user info
    }
}
